### PR TITLE
Update mechanism for funding accounts

### LINKED ISSUE #1277: 🚧 Dev ➾ MVP for Funding Default Testnet Account
### 🚥 Definition of Done

- [x] 🎯 Planning and Design
- [ ] 🏎️ Dev Implementation
- [ ] 🏁 Test Plan Executed
- [ ] 🛩️ Documentation Updated
- [ ] 🏟️ Peer / User Reviewed
- [ ] 🏆 Validated and Accepted

### ✅ To Do

_No response_

### 🔭 Scope

🏎️ In Scope: 
- Add a warning for the user if a faucet is detected in config.json
- Create the means to instantiate a root account (used for funding, originate, and transfer)
- Provide the UX to guide the user through the steps needed to fund the account
- Refactoring the `taq originate` and `taq transfer` tasks to use the root account
 
 🙅‍♂️ Out of Scope: 
- Any signing other than secretKey 
- Refactoring network/accounts/config
- Any work on accounts in a sandbox env  
- Removing faucet from the protocol or CORE
- `taq list accounts`, `taq fund accounts`, and `taq generate accounts`

### 🧰 Work Description

This ticket covers the MVP of the replacement for faucets on a testnet

MVP UX Flow:
1. User tries to originate to a testnet which does not have any accounts defined in the env
2. Taquito plugin/Taqueria will instantiate each of the defined accounts inside the relevant environment
3. Taquito will generate a keypair for each account and add that information to config.json
4. The CLI will output the newly generated accounts and their PHK's with instructions on how to fund the accounts (link to teztnets site and what they need to copy/paste)

### ⚖️ Acceptance Criteria

✨ Features: 
- [ ] A warning is provided to a user if a faucet has been defined
- [ ] Taqueria will disregard the faucet and use the root account per env for testnet secrets
- [ ] The means to generate defined accounts (and their keys) in a testnet environment
- [ ] A user flow to provide the PHK for each created account and instructions on how the user can fund them
- [ ] The root account will be set as the default account (for use in origination)
 
 📚 Documentation: 
- [ ] Updates to existing documentation regarding testnets/faucets
 
### 🎢 Test Plan

The initial PR should include basic e2e tests, but should keep in mind that this MVP will be subject to frequent and possibly dramatic changes and as such, the number of tests and work involved should be minimal

### 🎆 Release Notes

_No response_

### 🦄 Sizing

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by jchenche**
# 🏗️ PR ➾

Fixes #1277

## 🪂 Pre-Merge Checklist

- [x] 🐬 I have commented my code, particularly in hard-to-understand areas
- [x] 🦀 Automated tests have been written and added to this PR
- [x] 🤿 Corresponding changes have been made to all documentation
- [x] 🏖️ New and existing unit tests pass locally and in CI
- [x] 🔱 The test plan has been implemented and verified by an SDET
- [x] 🏄‍♂️ Myself and others have built this branch and done manual testing on the change
- [x] ⛵ A summary describing the change has been written for release notes

----------------------------------------------------------------------------------------------------------------------------

## 🪁 Description

To accommodate recent changes on how to fund accounts via faucets in testnets. We'll generate a keypair for you and ask you to go to https://teztnets.xyz to fund it. Then you can use it to fund other accounts or use it for operations such as origination and transfer.

## 🎢 Test Plan

Please describe the testing strategy for this change

----------------------------------------------------------------------------------------------------------------------------

### 🛸 Type of Change

- [ ] 🐟 Minor change (non-breaking change of very limited scope)
- [ ] 🦑 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🌊 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🐳 Major refactor (breaking changes and significant impact to the codebase)

----------------------------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Review by hu3man — APPROVED** [review]
Excellent work! The UX is great for an MVP and it dramatically improves on the faucet file flow

Tests are still failing it looks like but the code and CLI output both look good to me 👍

I do want to have the teztnets link clickable, but that can come in a later iteration

**Review by AlirezaHaghshenas — APPROVED** [review]

File: taqueria-plugin-taquito/transfer.ts [hunk 1]
```diff
@@ -122,7 +115,20 @@ const performTransferOp = (tezos: TezosToolkit, contractInfo: TableRow): Promise
...
 		.then(op => op.confirmation().then(() => op.hash))
-		.catch(err => sendAsyncErr(`Error during transfer operation:\n${err} ${JSON.stringify(err, null, 2)}`));
+		.catch(err => {
+			if (err instanceof Error) {
+				if (/empty_implicit_contract/.test(err.message)) {
+					const result = (err.message).match(/(?<="implicit":")tz[^"]+(?=")/);
+					const publicKeyHash = result ? result[0] : undefined;
+					if (publicKeyHash) {
+						return sendAsyncErr(
+							`${publicKeyHash} possibly has no funds. Go to https://teztnets.xyz and click "Faucet" of the desired protocol to fund it. Note that you might need to wait for a few seconds for the network to register the funds.`,
```
Comment (hu3man) [inline taqueria-plugin-taquito/transfer.ts:125]: The root account {publicKeyHask} for the target environment may not be funded \n
To fund this account:
1. Go to https://teztnets.xyz and click "Faucet" of the target testnet
2. Copy and paste {publicKeyHash} into the 'wallet address field
3. Request some Tez
(Note that you might need to wait for a few seconds for the network to register the funds)

File: taqueria-sdk/index.ts [hunk 2]
```diff
@@ -490,6 +495,43 @@ export const getAddressOfAlias = async (
...
+		network.accounts[account] = { publicKey, publicKeyHash, privateKey };
+		try {
+			await writeJsonFile('./.taq/config.json')(parsedArgs.config);
+		} catch (err) {
+			return sendAsyncErr(`Could not write to ./.taq/config.json\n`);
+		}
+		return sendAsyncErr(
+			`A keypair with public key hash ${
+				network.accounts[account].publicKeyHash
+			} was generated for you. Go to https://teztnets.xyz and click "Faucet" of the desired protocol to fund it.`,
```
Comment (hu3man) [inline taqueria-sdk/index.ts:529]: Same instructions as above

**Review by hu3man — CHANGES_REQUESTED** [review]

**Review by hu3man — APPROVED** [review]

**Comment by hu3man** [comment]
Still need doc updates right?

**Comment by jchenche** [comment]
> Still need doc updates right?

@hu3man I made the updates as part of the ghostnet addition.

**Review by hu3man — APPROVED** [review]

**Pull request merged**